System/InputScheme.h

```diff
@@ -93,11 +152,27 @@ namespace RTE {
 		void SetJoystickDeadzoneType(DeadZoneType deadzoneType) { m_JoystickDeadzoneType = deadzoneType; }
 #pragma endregion
 
-#pragma region Concrete Methods
+#pragma region Input Mapping Capture Handling
+		/// <summary>
+		/// Clears all mappings for a specific input element of a specific player.
+		/// </summary>
+		/// <param name="whichInput">Which input element to clear all mappings of.</param>
+		void ClearMapping(int whichInput) { m_InputMappings.at(whichInput).Reset(); }
+
+		/// <summary>
+		/// Checks for any key press this frame and creates an input mapping for a specific player accordingly.
+		/// </summary>
+		/// <param name="whichInput">Which input element to map to for that player.</param>
+		/// <returns>Whether there were any key presses this frame and therefore whether a mapping was successfully captured or not.</returns>
+		bool CaptureKeyMapping(int whichInput);
+
 		/// <summary>
-		/// Sets some sensible default bindings for this.
+		/// Checks for any button or direction press this frame and creates an input mapping for a specific player accordingly.
```

Comment:
Joystick seems to be a pretty universal name for controllers. I guess it's technically correct.
I can change it to `Gamepad` everywhere sometime later.

Reply:
I like gamepad the most out of the 3 options, though I didn't realize it was technically incorrect lol. I guess I'm okay with either joystick or gamepad, as long as we try to stick with one. As I mentioned in some other comment, I definitely wanna stop calling it (and thinking of it as) controller so there's no stupid naming confusion with the `Controller` class.

Reply:
The lines are kinda blurry now, there's no real difference.
But back in the day gamepads had buttons and joysticks had analogs + less buttons.
It all got mixed up when analogs were introduced into what were gamepads, e.g the N64 controller and later DualShock.

I guess the most proper name for them would be `InputDeviceWhichIsNotOptimalForThisGame`. ¯\\\_( ͡° ͜ʖ ͡°)_/¯

Reply:
hahaha we can only call it that if we show that to users in the input customization menus :P

Anyway, I'll leave the decision on what name is best to you, since this is definitely more your wheelhouse than mine. And feel free to do it as another PR instead of this one, I don't mind.